### PR TITLE
[DF] Rename RDF factory functions

### PR DESCRIPTION
**Pull request by eguiraud**
This PR changes names such as `ROOT::RDF::MakeCsvDataFrame` to the nicer/clearer convention `ROOT::RDF::FromCSV`, deprecating the previous spellings.

Although what's nicer and clearer is generally subjective, I think there is strong enough evidence in this case to warrant the change: 

- https://root-forum.cern.ch/t/makenumpydataframe-function-name/50566
- the `FromXXX` naming convention is already widespread across libraries and programming languages (e.g. [rust naming conventions](https://rust-lang.github.io/api-guidelines/naming.html#naming), [pandas](https://pandas.pydata.org/pandas-docs/stable/reference/frame.html#serialization-io-conversion)
- the current naming convention does not work for factory functions that we might want in the future such as `ROOT::RDF::FromJSONDatasetSpec` (`MakeJSONDatasetSpecDataFrame`, `MakeJSONDataFrame`?)

### To do

- [x] check with @jblomer that moving `ROOT::Experimental::MakeNTupleDataFrame` to `ROOT::RDF::Experimental::FromNTuple` (not the change of namespace) is ok
- [x] rename `ROOT.RDF.MakeNumpyDataFrame` to `ROOT.RDF.FromNumpy`: @etejedor how do I deprecate a Python-only method?
- [x] `FromCSV` vs `FromCsv` -- I think it should be `CSV`, just like we already have [`TBufferJSON::FromJSON`](https://root.cern/doc/master/classTBufferJSON.html#abefb8455a4514bc7ea5e722ebd2dc203)

P.S.
in case it wasn't clear, this is basically a RFC :)

### CODE REVIEW

**Comment by etejedor** [comment]
>  rename ROOT.RDF.MakeNumpyDataFrame to ROOT.RDF.FromNumpy: @etejedor how do I deprecate a Python-only method?

We can do the same as for the deprecation of `TTree.AsMatrix`, see [commit](https://github.com/root-project/root/commit/8b7ba03bf8fb843b018ea38b7c7cd163480e7a51#), which mainly consists in issuing a Python [FutureWarning](https://docs.python.org/3/library/exceptions.html#FutureWarning) for the old API during one release.

**Comment by ikabadzhov** [comment]
I agree that `FromCSV` should be more consistent with our current api.

**Comment by eguiraud** [comment]
Thanks Enric, I took a quick look and there is a number of places that need updating in the bindings, let's take a look together when you are back :)

**Comment by jblomer** [comment]
While we are at it, should we call it `ROOT::RDF::Experimental::FromRNTuple` to avoid confusion with `TNtuple`

File: tree/dataframe/inc/ROOT/RSqliteDS.hxx [hunk 1]
```diff
@@ -113,6 +113,9 @@ protected:
    Record_t GetColumnReadersImpl(std::string_view name, const std::type_info &) final;
 };
 
+RDataFrame FromSqlite(std::string_view fileName, std::string_view query);
```
Comment (vepadulano) [inline tree/dataframe/inc/ROOT/RSqliteDS.hxx:116]: I don't know whether we should take the chance to conform to its official naming: https://www.sqlite.org/index.html

**Review by vepadulano — COMMENTED** [review]
Overall I think this makes a lot of sense! And it is also more future-proof as you discussed. My only doubt is the semantical difference between a factory that creates an RDF from a dataset in some format (e.g. `FromCSV`) and a factory that creates an RDF from a specification (e.g. `FromJSON`). We can discuss whether appending `Spec` is ok for the latter to be more explicit, if it's needed at all, or if we want something else

File: tree/dataframe/inc/ROOT/RSqliteDS.hxx (hunk 1 above)
Comment (eguiraud) [reply tree/dataframe/inc/ROOT/RSqliteDS.hxx:115]: absolutely, thank you

Comment (eguiraud) [reply tree/dataframe/inc/ROOT/RSqliteDS.hxx:115]: ah, there is the incredibly annoying problem that the header to include is called `RSqliteDS.hxx` :man_facepalming:

**Comment by eguiraud** [comment]
Need to fix Python 2 and change the capitalization of `RSqliteDS.hxx` to `RSQLiteDS.hxx`

**Comment by eguiraud** [comment]
@vepadulano I tried to get `FromSQLite` instead of `FromSqlite` but that would kind of require that we also rename `RSqliteDS.hxx` to `RSQLiteDS.hxx` and MacOS has a case-insensitive file-system which makes that really hard (because I'd like to have _both_ spellings in, deprecating the old one, but MacOS doesn't like that at all).

I'd propose to address FromSqlite/FromSQLite in a different PR.

**Review by vepadulano — APPROVED** [review]
> MacOS has a case-insensitive file-system which makes that really hard 

Good old MacOS :smile: I understand, fine for me

**Comment by jblomer** [comment]
@eguiraud Regarding `ROOT::RDF::Experimental::FromNTuple`: looks good to me!

**Comment by eguiraud** [comment]
@phsft-bot build just on windows10/cxx14

**Comment by eguiraud** [comment]
Windows failures are unrelated. Merging.

**Pull request merged**